ledger/conway/conway.go

```diff
@@ -75,12 +75,43 @@ func (b *ConwayBlock) UnmarshalCBOR(cborData []byte) error {
 		return err
 	}
 
-	// Convert []any to []uint
+	// Convert []any to []uint with validation
+	const maxReasonableIndex = 1000000 // Reasonable upper bound for transaction indices
+	result := make([]uint, 0, len(tmp.InvalidTransactions))
+	for _, v := range tmp.InvalidTransactions {
+		switch val := v.(type) {
+		case uint:
+			if val > maxReasonableIndex {
+				continue // Skip unreasonably large indices
+			}
+			result = append(result, val)
+		case uint64:
+			if val > maxReasonableIndex {
+				continue // Skip unreasonably large indices
+			}
+			result = append(result, uint(val))
+		case int:
+			if val < 0 || val > maxReasonableIndex {
+				continue // Skip negative or unreasonably large indices
+			}
+			result = append(result, uint(val))
+		case int64:
+			if val < 0 || val > maxReasonableIndex {
+				continue // Skip negative or unreasonably large indices
+			}
+			result = append(result, uint(val))
+		default:
+			// Skip invalid types (strings, floats, etc.)
+			continue
+		}
+	}
+	b.InvalidTransactions = result
```

Comment:
There is no reason to be decoding to `[]any` in the first place. You only need the `[]any` when encoding/marshaling, and only for `cbor.IndefLengthList`